### PR TITLE
Add Python 3.5 compatibility

### PR DESCRIPTION
**Pull request by pawroman**
Add Python 3.5 compatibility. All the tests should pass.

- [ ] Code conforms to style guide
- [ ] Doctests
- [ ] Unit-tests
- [x] Documentation
- [ ] New line item in [releases.md](releases.md)
- [ ] Info added to `literature` directory

### CODE REVIEW

**Review by dwillmer — COMMENTED** [review]
LGTM

**Comment by dwillmer** [comment]
I'm not too bothered about 3.5 support, but if you think it's useful then we'll merge it.

Nice one!

File: tests/data/datasets.py [hunk 1]
```diff
@@ -4,9 +4,9 @@
...
-    DIABETES = datasets.load_diabetes()
-    LINNERUD = datasets.load_linnerud()
-    IRIS = datasets.load_iris()
-    WINE = datasets.load_wine()
+    BOSTON = datasets.load_boston().data
+    CANCER = datasets.load_breast_cancer().data
+    DIABETES = datasets.load_diabetes().data
+    LINNERUD = datasets.load_linnerud().data
+    IRIS = datasets.load_iris().data
+    WINE = datasets.load_wine().data
```
Comment (pawroman) [inline tests/data/datasets.py:12]: Quite a weird issue. Python 3.5's Enum doesn't like sklearn's `Bunch` class. This workaround gets rid of random test failure (at least on my machine).

**Comment by pawroman** [comment]
Yeah, not supporting Python 3.5 feels like an artificial limitation. The only issue I could find was the `Enum` not liking sklearn's `Bunch` dataset container. Other than that, all good.

3.5 also widens our audience (corporates etc).

**Comment by pawroman** [comment]
That being said, I see no reason to drop 3.5 support if maintaining it becomes a hassle.

**Review by dwillmer — APPROVED** [review]
LGTM!

**Pull request merged**